APMrover2/AP_MotorsUGV.cpp

```diff
@@ -414,12 +408,20 @@ bool AP_MotorsUGV::pre_arm_check(bool report) const
         return false;
     }
     // check if only one of throttle or steering outputs has been configured, if has a sail allow no throttle
-    if ((has_sail() || SRV_Channels::function_assigned(SRV_Channel::k_throttle)) != SRV_Channels::function_assigned(SRV_Channel::k_steering)) {
+    if ((rover.get_sailboat_enable() || SRV_Channels::function_assigned(SRV_Channel::k_throttle)) != SRV_Channels::function_assigned(SRV_Channel::k_steering)) {
```

Comment:
I'm not a fan of adding more dependencies on the rover class from the lower level AP_MotorsUGV class.  I was actually planning to move AP_MotorsUGV to it's own directory to harden the interface.  So perhaps it seems like duplication but I'd prefer if the has_sail() could remain (perhaps it can be made private) and have the AP_MotorsUGV class keep using it.

Reply:
I have dropped the the AP_MotorsUGV changes and moved has_sail() to protected.